src/app/shared/editor/services/monaco.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from "@angular/core";
+import * as monaco from "monaco-editor";
+import { MaybeNull } from "../../../common/app.types";
+/* eslint-disable */
+export function getMonacoNamespace(): typeof monaco {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+    return (window as any).monaco as typeof monaco;
+}
+
+@Injectable({
+    providedIn: "root",
+})
+export class MonacoService {
+    public setErrorMarker(model: monaco.editor.ITextModel, error: MaybeNull<string>): void {
+        const monaco = getMonacoNamespace();
+
+        if (!monaco) return;
+
+        if (error) {
+            const markerData = this.prepareMarkerData(error);
+            monaco.editor.setModelMarkers(model, "", [markerData]);
+        } else {
+            monaco.editor.setModelMarkers(model, "", []);
+        }
+    }
+
+    private prepareMarkerData(error: string): monaco.editor.IMarkerData {
+        const { line, col } = this.getErrorPos(error);
+
+        return {
+            startLineNumber: line || 1,
+            startColumn: col || 1,
+            endLineNumber: line || 9999,
```

Comment:
Can't we extract number of lines and number of columns in last line from the editor's model?

Reply:
Yes, we can, thank you, changed